tests/test_linkextractors.py

```diff
@@ -851,3 +852,36 @@ def test_skip_bad_links(self):
                 ),
             ],
         )
+
+    def test_link_allowed_is_false_with_empty_url(self):
+        mock_link = Mock()
+        mock_link.url = ""
+        expected = False
+
+        actual = LxmlLinkExtractor()._link_allowed(mock_link)
+
+        self.assertEqual(expected, actual)
+
+    def test_link_allowed_is_false_with_bad_url_prefix(self):
+        mock_link = Mock()
+        mock_link.url = "htp://should_be_http.com"
```

Comment:
`should_be_http.com` could become a real domain in the future. Please use a domain ending in `.example` (instead of `.com`) when needing a custom domain name.